client/client.go

```diff
@@ -116,6 +116,17 @@ type SysInfo struct {
 	Store            string `json:"store,omitempty"`
 }
 
+// processErrorResponse handles the common error path for API requests.
+// This function should be called when response.Type == "error"
+func (rsp *response) processErrorResponse() error {
```

Comment:
Can we please call this method simply `err() error`, and make it return nil if response.Type != nil?

Suggested prototype:
```
// err returns the error represented by the response, or nil if it's not an error.
func (rsp *response) err() error {
```
This allows using the traditional convention:
```
if err := rsp.err(); err != nil {
        return err
}
```

For the record, Err is the common name for methods that return the error observed by some value. Since this is private, err does the trick.